packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -376,25 +376,33 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		);
 	}
 
-	private getBlobHandle(id: string): BlobHandle {
+	private getBlobHandle(localId: string, storageId: string | undefined): BlobHandle {
 		assert(
-			this.redirectTable.has(id) || this.pendingBlobs.has(id),
+			this.redirectTable.has(localId) || this.pendingBlobs.has(localId),
 			0x384 /* requesting handle for unknown blob */,
 		);
-		const pending = this.pendingBlobs.get(id);
+		const pending = this.pendingBlobs.get(localId);
 		const callback = pending
 			? () => {
 					pending.attached = true;
 					this.emit("blobAttached", pending);
-					this.deletePendingBlobMaybe(id);
+					this.deletePendingBlobMaybe(localId);
 			  }
 			: undefined;
-		return new BlobHandle(
-			`${BlobManager.basePath}/${id}`,
-			this.routeContext,
-			async () => this.getBlob(id),
-			callback,
-		);
+
+		return storageId
+			? new BlobHandle(
+					`${BlobManager.basePath}/${storageId}`,
```

Comment:
this seems wrong. we never validate the storage id is valid like we do the local id

Reply:
Not sure what you mean by we never validate storage id